Modules/xASL_module_ASL.m

```diff
@@ -190,6 +191,54 @@
 	x.Q.NumberEchoTimes = 1;
 end
 
+%% D6. Automatic session merging
+
+% SessionMergingList == total list of sessions that need concatenating (can be multiple lists for multiple concatenations)
+% Read the list of sessions to merge, defaulting to empty
+if ~isfield(x.modules.asl, 'SessionMergingList')
+	x.modules.asl.SessionMergingList = {};
+else
+	% The parameter should be a list of lists, but if a single list is provided, it converts it
+	if ~isempty(x.modules.asl.SessionMergingList) && iscell(x.modules.asl.SessionMergingList) && ~iscell(x.modules.asl.SessionMergingList{1})
+		x.modules.asl.SessionMergingList = {x.modules.asl.SessionMergingList};
+    end
+    nLists = numel(x.modules.asl.SessionMergingList);
+    fprintf('\n%s', ['-> Detected ' xASL_num2str(nLists) ' list(s) for concatenating sessions']);
+end
+
+x.modules.asl.bMergingSessions = 0; % By default, we do not merge any sessions
+% sessionsToMerge = session list that we will concatenate for this specific session & run of xASL_module_ASL
+% e.g., if we want to concatenate ASL_1 & ASL_2, this will only contain both sessions for the xASL_module_ASL iteration of ASL_2
+x.modules.asl.sessionsToMerge = {};
+
+
+% Find the lists containing the current session
+% Note that these can be multiple lists
+for iList=1:nLists
+	if ~isempty(x.modules.asl.SessionMergingList{iList}) && sum(ismember(x.SESSION, x.modules.asl.SessionMergingList{iList}))
+		x.modules.asl.sessionsToMerge = x.modules.asl.SessionMergingList{iList};
+
+        if ~isempty(x.modules.asl.sessionsToMerge) && strcmp(x.SESSION, x.modules.asl.sessionsToMerge{end})
```

Comment:
Here you see I simplified the code a bit, it didn't matter if `x.modules.asl.sessionsToMerge` is inside or outside the loop.

One thing that I don't understand yet, is that we allow multiple merging lists (the cell in cell structure you enforce above), but we don't deal with them properly. E.g., if we have merging lists:

ASL_1 ASL_3
ASL_2 ASL_3

then for ASL_3, we will only concatenate one of these lists in the current situation (irregardless of your version or this new version). I don't think we should already allow multiple mergings with the same last session, because that makes our life tricky, but we should at least issue a warning don't you think?

Reply:
Good, but:
1. The reason for allowing several lists is that you can have a DEBBIE reproducibility study for example. You are right that we are concatenating lists for the last session and I definitively agree that multiple mergins with the same last session should never happen - I just didn't think of that happening. So we should never have two lists with the same end sessions. So 1+3 and 1+4 would be allowed. But 1+3 and 2+3 will not be allowed. I will simply write a function that checks this and then takes only the first session list and reports a warning. Or something like that.
2. Your code will crash on line 217. nList is not initialized if the list was not existing on line 197.
3. Related to point 1. You are right about this and we have to add an additional check. However, we already select the correct list on line 219. So line 223 has to be deleted.
4. Line 234 - you can simplify the code. for in Matlab works really nice for this
Instead of "n = numel(list), for i=1:n, print(list(i))" you can simply write for i=list, print(i) - that does exactly the same and is easier to read.

Reply:
1. I agree, then shall we put that here then? I would write that already now, it won't take that much extra time.
2. Added nLists = 0 as dummy in 06081dd879b63b008be821d6baabc965cda9e5e5
3. We can't remove this because we need to check for x.SESSION being the latest index. I have now removed the first check which was unnecessary in 665dadb499182ba4edb67cd3fc57918492e0a51f
4. That's only the case in your example if list is already numerical right? In the example in this code, the list is not numerical but multiple cells containing ASL_1 ASL_2 etc. Otherwise, this would definitely not be easier to read (it would not make sense to make a for loop with string-cells as indices...

BTW: I'm happy for you to fix such things of course, I just wanted to simplify your code when I was trying to make it more readable by adding comments

Reply:
1. Yes - we that already now. I'll do it, just waiting for you to agree.
2. Solved.
3. Yes, the first check wasn't necessary. But Line 229 (that was recently added) is identical to line 225 and should be removed.
4. Parameter `list2concatenate` should be removed as on line 238 and 240 we should be using `x.modules.asl.sessionsToMerge` instead. This works also for cells. Test this:
`c{1} = 'a'; c{2} = 'bc';for i=c,fprintf(['test ' i{1} '\n']),end`
You don't use string-cells as indices, you don't use indices at all. For simply goes across any array. 1:N is an array of integers, so you can have an array of strings - the exact same mechanism and you skip indexing completely.

Reply:
3. Done in 69be7466ffa370657a2a5f3b23d7f21acef92fd2
4. Done in c08d02a902a9e99831f74d7a1d009b8f49cc8902

for goes across a list of indices, either internally (non-explicit to the user) or specified directly (clear to the user). Your example saves 1 line of code but makes it less easy readable. So I would keep it like we have it. Unless we want a new discussion again ;)

`c{1} = 'a'; c{2} = 'bc'; nI = numel(c); for i=1:nI,fprintf(['test ' i{1} '\n']),end`
will just cost a few words extra, but is more explicit. So I like it more. Note also that this behavior might vary between programming languages right?
BTW: note that `i` is not a good variable right because you can mistake it for the imaginary number, that's why we always use at least ii? Please don't always try to look at things from the ex-mathematical/physics student perspective ;)

Reply:
I will still program the check for extra sequences - point 1.

I'm fine keeping it as it is as we removed the extra redundant code, though it's still better to add `numel` directly inside `for` without creating more variable.

Reply:
Check done in 006d5958
I take the first eligible session. Issue a warning when a session fits in multiple lists. And a warning when at multiple endings. But a list ending with a current session has a priority.